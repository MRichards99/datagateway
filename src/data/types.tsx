export interface InvestigationData {
  ID: string;
  TITLE: string;
  VISIT_ID: number;
  RB_NUMBER: string;
  DOI: string;
  SIZE: number;
  INSTRUMENT: { NAME: string };
  STARTDATE: Date;
  ENDDATE: Date;
  [key: string]: string | number | Date | { NAME: string };
}

export interface DatasetData {
  ID: string;
  NAME: string;
  SIZE: number;
  CREATE_TIME: Date;
  MOD_TIME: Date;
  [key: string]: string | number | Date;
}

export interface DatafileData {
  ID: string;
  NAME: string;
  LOCATION: string;
  SIZE: number;
  MOD_TIME: Date;
  [key: string]: string | number | Date;
}

export interface FoodData {
  id: number;
  dessert: string;
  calories: number;
  fat: number;
  carbs: number;
  protein: number;
  [key: string]: string | number;
}

export type EntityType =
  | InvestigationData
  | DatasetData
  | DatafileData
  | FoodData;
