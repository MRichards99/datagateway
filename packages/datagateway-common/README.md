# datagateway-common

A library containint common code and components for the DataGateway project.
It is a dependency of other packages in the repo and other apps can import code from it
like normal.

When adding new components/functions/definitions/constants etc., you must export the item
in `index.tsx` so that it will be importable from other applications.

e.g.

In `some_file.tsx`

```
export const my_const = 1;
```

In `index.tsx`

```
export { my_const } from 'some_file.tsx'
export { my_const };
```

If you export a lot of items in `some_file.tsx` and want to export all of them to be exposed to other packages
then the following syntax can be used instead

`index.tsx`

```
export * from 'some_file.tsx'
```

## Available Scripts

In the project directory, you can run:

### `npm install`

Installs dependencies for `datagateway-common` only. Should normally use `npx lerna add` at the
top level, but can use regular `npm install` if it's only used by `datagateway-common` and not other
packages.

### `npm start`

Runs the app in development mode
Open [http://localhost:3000](http://localhost:3000) to view it in the browser.

In order to begin local development, you should uncomment the lines in `index.tsx`.
These allow the app to load like a regular React app rather than behave as a library.

### `npm run tsc`

Runs the Typescript compiler that converts our Typescript files into a Javascript library
that can be imported by our other packages. When making changes to this app, you must run
this to see the changes reflected in other apps. Think of it as a build step like `npm run build`.

### `npm test`

Runs unit tests

### `npm run test:watch`

Runs unit tests in watch mode, allowing you to filter which tests to run and automatically restart tests on code changes.

### `npm run lint:js`

Lints the `src` folder
