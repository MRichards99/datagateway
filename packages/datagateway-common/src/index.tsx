// Uncomment to use App to test components
// import React from 'react';
// import ReactDOM from 'react-dom';
// import App from './App';

export * from './app.types';
export * from './state/actions/actions.types';

export { default as Table } from './table/table.component';
export * from './table/table.component';
export { default as DetailsPanelRow } from './table/rowRenderers/detailsPanelRow.component';
export { default as DataHeader } from './table/headerRenderers/dataHeader.component';
export { default as TextColumnFilter } from './table/columnFilters/textColumnFilter.component';
export { default as DateColumnFilter } from './table/columnFilters/dateColumnFilter.component';
export { default as ActionCellComponent } from './table/cellRenderers/actionCell.component';
export { default as DataCellComponent } from './table/cellRenderers/dataCell.component';
export { default as ExpandCellComponent } from './table/cellRenderers/expandCell.component';
export * from './table/cellRenderers/cellContentRenderers';
export * from './state/actions/datafiles';
export * from './state/actions/datasets';
export * from './state/actions/investigations';
export * from './state/actions/cart';
export * from './state/actions/instruments';
export * from './state/actions/facilityCycles';
export { default as Preloader } from './preloader/preloader.component';
export { default as dGCommonReducer } from './state/reducers/dgcommon.reducer';

import { StateType } from './state/app.types';
export type DGCommonState = StateType;

// ReactDOM.render(<App />, document.getElementById('root'));
