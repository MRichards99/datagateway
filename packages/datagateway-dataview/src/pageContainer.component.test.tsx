import React from 'react';
import { ReactWrapper } from 'enzyme';

import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { StateType } from './state/app.types';
import { initialState as dgDataViewInitialState } from './state/reducers/dgdataview.reducer';
import { dGCommonInitialState } from 'datagateway-common';

import { createShallow } from '@material-ui/core/test-utils';
// history package is part of react-router, which we depend on
// eslint-disable-next-line import/no-extraneous-dependencies
import { createLocation } from 'history';
import { MemoryRouter } from 'react-router';

import PageContainer from './pageContainer.component';
// import { I18nextProvider } from 'react-i18next';
// import i18n from 'i18next';
// import { initReactI18next } from 'react-i18next';
// i18n.use(initReactI18next).init({
//   lng: 'en',
//   fallbackLng: 'en',
//   // have a common namespace used around the full app
//   ns: ['translations'],
//   defaultNS: 'translations',
//   debug: true,
//   interpolation: {
//     escapeValue: false, // not needed for react!!
//   },
//   resources: { en: { translations: {} } },
// });

jest.mock('loglevel');

describe('PageContainer - Tests', () => {
  let shallow;
  let state: StateType;

  const createWrapper = (state: StateType): ReactWrapper => {
    const mockStore = configureStore([thunk]);
    return shallow(
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <PageContainer store={mockStore(state)} />
      </MemoryRouter>
      // {
      //   wrappingComponent: I18nextProvider,
      //   wrappingComponentProps: {
      //     i18n,
      //   },
      // }
    );
  };

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'Grid' });

    state = JSON.parse(
      JSON.stringify({
        dgcommon: { ...dGCommonInitialState, totalDataCount: 101 },
        dgdataview: dgDataViewInitialState,

        router: {
          action: 'POP',
          location: createLocation('/'),
        },
      })
    );
  });

  it('displays the correct entity count', () => {
    // Set up store with the test state and mounted page head.
    const wrapper = createWrapper(state);

    expect(wrapper).toMatchSnapshot();
  });
});
