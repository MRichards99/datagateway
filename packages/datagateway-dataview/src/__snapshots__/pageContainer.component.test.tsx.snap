// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer - Tests displays the correct entity count 1`] = `
<div
  className="MuiGrid-root WithStyles(ForwardRef(Grid))-root-1 MuiGrid-container"
>
  <WithStyles(ForwardRef(Grid))
    aria-label="container-breadcrumbs"
    item={true}
    xs={true}
  >
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(TranslatedBreadcrumbs)",
          "type": [Function],
        }
      }
      path="/browse"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    aria-label="container-table-count"
    item={true}
    style={
      Object {
        "textAlign": "center",
      }
    }
    xs={2}
  >
    <WithStyles(ForwardRef(Paper))
      square={true}
      style={
        Object {
          "backgroundColor": "inherit",
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <WithStyles(ForwardRef(Typography))
        component="h3"
        variant="h6"
      >
        <b>
          app.results
          :
        </b>
         
        101
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Paper))
    square={true}
    style={
      Object {
        "backgroundColor": "inherit",
        "display": "flex",
      }
    }
  >
    <WithStyles(ForwardRef(IconButton))
      aria-label="container-table-search"
      onClick={[Function]}
      style={
        Object {
          "margin": "auto",
        }
      }
    >
      <SearchIcon />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(Paper))
    square={true}
    style={
      Object {
        "backgroundColor": "inherit",
        "display": "flex",
      }
    }
  >
    <WithStyles(ForwardRef(IconButton))
      aria-label="container-table-cart"
      onClick={[Function]}
      style={
        Object {
          "margin": "auto",
        }
      }
    >
      <WithStyles(ForwardRef(Badge))
        aria-label="container-table-cart-badge"
        badgeContent={null}
        color="primary"
      >
        <ShoppingCartIcon />
      </WithStyles(ForwardRef(Badge))>
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(Grid))
    aria-label="container-table"
    item={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Paper))
      square={true}
      style={
        Object {
          "backgroundColor": "inherit",
          "height": "calc(100vh - 112px)",
          "width": "100%",
        }
      }
    >
      <PageTable />
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
